src/server.ts

```diff
@@ -78,7 +78,7 @@ export type SubscribeFunction = (schema: GraphQLSchema,
 export interface ServerOptions {
   rootValue?: any;
   schema?: GraphQLSchema;
-  execute?: ExecuteFunction;
+  execute?: Promise<ExecuteFunction> | ExecuteFunction;
```

Comment:
I don't feel that this is the correct place to put the optional Promise as this could mean you are getting a Promise of a Promise which isn't quite right. This did inspire a different fix which seems to be working and will push out as 0.9.9.